mars/services/scheduling/worker/workerslot.py

```diff
@@ -151,6 +154,16 @@ def release_free_slot(self, slot_id: int, pid: Optional[int] = None):
             event = self._slot_kill_events.pop(slot_id)
             event.set()
 
+        if pid is not None:
+            if slot_id in self._slot_to_session_stid:
+                # We should release the slot by one role, if the slot is
+                # acquired by the SubtaskExecutionActor, then the slot
+                # should be released by it, too.
+                session_stid = self._slot_to_session_stid[slot_id]
+                logger.info('Slot %s released by pid %s, but not for %s',
+                            slot_id, pid, session_stid)
+                return
```

Comment:
This section of code is a little confusing here. I would prefer adding a separate actor API maintaining slot-pid bindings.

Reply:
I prefer adding a `register_slot` API for the `BandSlotControlActor`.

Reply:
I have added a `register_free_slot` API for the `BandSlotControlActor`.